lib/api/CAnomalyJobConfig.cc

```diff
@@ -347,17 +366,108 @@ bool CAnomalyJobConfig::CAnalysisConfig::processFilter(const std::string& key,
+    const boost::property_tree::ptree& propTree,
+    const std::string& key,
+    const std::string& value,
+    TIntSet& handledScheduledEvents) {
+    // Here we pull out the "1" in "scheduledevent.1.description"
+    // description may contain a '.'
+    std::size_t sepPos{key.find(SUFFIX_SEPARATOR, SCHEDULED_EVENT_PREFIX.length() + 1)};
+    if (sepPos == std::string::npos || sepPos == key.length() - 1) {
+        LOG_ERROR(<< "Unrecognised configuration option " << key << " = " << value);
+        return false;
+    }
+
+    std::string indexString{key, SCHEDULED_EVENT_PREFIX.length(),
+                            sepPos - SCHEDULED_EVENT_PREFIX.length()};
+    int indexKey;
+    if (core::CStringUtils::stringToType(indexString, indexKey) == false) {
+        LOG_ERROR(<< "Cannot convert config key to integer: " << indexString);
+        return false;
+    }
+
+    // Check if we've already seen this key
+    if (handledScheduledEvents.insert(indexKey).second == false) {
+        // Not an error
+        return true;
+    }
+
+    std::string description{propTree.get(
+        boost::property_tree::ptree::path_type{
+            SCHEDULED_EVENT_PREFIX + indexString + DESCRIPTION_SUFFIX, '\t'},
+        EMPTY_STRING)};
+
+    std::string rules{propTree.get(
+        boost::property_tree::ptree::path_type{
+            SCHEDULED_EVENT_PREFIX + indexString + RULES_SUFFIX, '\t'},
+        EMPTY_STRING)};
+
+    CDetectionRulesJsonParser::TDetectionRuleVec detectionRules;
+    if (this->parseRules(detectionRules, rules) == false) {
+        // parseRules() will have logged the error
+        return false;
+    }
+
+    if (detectionRules.size() != 1) {
+        LOG_ERROR(<< "Scheduled events must have exactly 1 rule");
+        return false;
+    }
+
+    m_ScheduledEvents.emplace_back(description, detectionRules[0]);
+
+    return true;
+}
+
+// TODO: Process updates as JSON
+bool CAnomalyJobConfig::CAnalysisConfig::parseRules(CDetectionRulesJsonParser::TDetectionRuleVec& detectionRules,
+                                                    const std::string& rules) {
+    if (rules.empty()) {
+        return true;
+    }
+
+    CDetectionRulesJsonParser rulesParser{m_RuleFilters};
+    return rulesParser.parseRules(rules, detectionRules);
+}
+
+ml::model::CAnomalyDetectorModelConfig
+CAnomalyJobConfig::CAnalysisConfig::makeModelConfig() const {
+    ml::model_t::ESummaryMode summaryMode{
+        m_SummaryCountFieldName.empty() ? ml::model_t::E_None : ml::model_t::E_Manual};
+
+    ml::model::CAnomalyDetectorModelConfig modelConfig{ml::model::CAnomalyDetectorModelConfig::defaultConfig(
+        m_BucketSpan, summaryMode, m_SummaryCountFieldName, m_Latency, m_MultivariateByFields)};
+
+    modelConfig.detectionRules(
+        ml::model::CAnomalyDetectorModelConfig::TIntDetectionRuleVecUMapCRef(m_DetectorRules));
+
+    modelConfig.scheduledEvents(
+        ml::model::CAnomalyDetectorModelConfig::TStrDetectionRulePrVecCRef(m_ScheduledEvents));
```

Comment:
The `ml::` qualifiers aren't needed here, as the whole class is in the `ml` namespace.